src/Core/Domain/Product/Command/UpdateProductCategoriesCommand.php

```diff
@@ -88,4 +87,32 @@ public function getCategoryIds(): array
     {
         return $this->categoryIds;
     }
+
+    /**
+     * @param int[] $categoryIds
+     */
+    private function setCategoryIds(array $categoryIds): void
+    {
+        $this->assertCategoryIdsAreNotEmpty($categoryIds);
```

Comment:
I think this assertion is not needed, because `new CategoryId($id)` should issue an exception if `$id` is empty.

You could catch the exception and re issue another one if needed

Reply:
If its empty array, the loop wont start and no exception will be thrown

Reply:
My mistake, I read too fast, for some reason I thought you were checking each element for non emptiness.